core/templates/App.ts

```diff
@@ -129,6 +129,8 @@ angular.module('oppia').config([
       $locationProvider.html5Mode(true);
     }
 
+    $sanitizeProvider.enableSvg(true);
```

Comment:
I am worried about this, and there are warnings on this page: https://docs.angularjs.org/api/ngSanitize/provider/$sanitizeProvider . How has this been handled? I also don't want us to be enabling SVG in HTML/other fields, and we already use SVG images elsewhere I think, so is this line actually necessary?

Reply:
Re the click-jacking issue mentioned on the page, I plan to add a general css rule to prevent overflow of content. Also, SVG files will be rendered inside a container that has the height and width specified so I don't really think this should be an issue ie. SVGs cannot be positioned outside their containers.

Reply:
Noted, but you'll need backend/lint checks and detailed explanation to ensure that that CSS rule never gets removed by a developer who may not understand the broader implications. If enableSvg() is part of this PR, then that fix should be too.

It's important (from a security perspective) to think about what could happen in the future, not just what could happen now. Currently SVGs are rendered inside containers. In the future they may not be, and the developers of such additional features might not realize the implications. So if possible try to avoid using enableSvg() as a global setting.

Reply:
I tried investigating this and it looks like enableSvg needs to be called during the "config" phase. Since we can't use a provider inside a factory/service. I tried creating a new provider where I set the enableSvg boolean, then I instantiated the sanitizeProvider and unset the boolean immediately. This seems to work. However, if we use $sanitize in the same directive along with the new provider it will allow SVG elements. This is because of the way AngularJS implements this. See below:

https://github.com/angular/angular.js/blob/418355f1cf9a9a9827ae81d257966e6acfb5623a/src/ngSanitize/sanitize.js#L147-L163

"validElements" gets extended with the SVG elements but it doesn't get removed if the flag is unset. Unless the provider is instantiated again, I guess.

I'm not sure how else to handle this (ie. setting the flag when necessary and unsetting it later). Is there any other way to do this in a contained manner withing the directive itself?

Reply:
This is what I tried in App.ts:

```ts
$provide.provider('SanitizeSVG', function() {
    return {
      $get: [
        '$injector', function($injector) {
          var obj = $injector.instantiate($sanitizeProvider.enableSvg(true).$get);
          $sanitizeProvider.enableSvg(false);
          return obj;
        }
      ]
    }
  });
```

Reply:
Chatted offline about this. One suggestion was to have a backend endpoint that can handle sanitization rather than using AngularJS's $sanitize service (which probably is a sensible thing to do since we are moving off AngularJS anyway).